Binary files partclone-0.2.69/get_lib_version and partclonebackup/get_lib_version differ
diff -rupN partclone-0.2.69/po/POTFILES partclonebackup/po/POTFILES
--- partclone-0.2.69/po/POTFILES	2013-12-26 02:16:44.000000000 -0500
+++ partclonebackup/po/POTFILES	1969-12-31 19:00:00.000000000 -0500
@@ -1,2 +0,0 @@
-     ../src/progress.c \
-     ../src/partclone.c
diff -rupN partclone-0.2.69/src/partclone.c partclonebackup/src/partclone.c
--- partclone-0.2.69/src/partclone.c	2013-12-26 02:16:44.000000000 -0500
+++ partclonebackup/src/partclone.c	2014-05-26 19:55:52.647730049 -0400
@@ -400,8 +400,8 @@ int open_ncurses() {
 #ifdef HAVE_LIBNCURSESW
 	int debug = 1;
 
-	FILE *in = fopen( "/dev/stderr", "r" );
-	FILE *out = fopen( "/dev/stderr", "w" );
+	FILE *in = fopen( "/dev/stdout", "r" );
+	FILE *out = fopen( "/dev/stdout", "w" );
 	int terminal_x = 0;
 	int terminal_y = 0;
 
diff -rupN partclone-0.2.69/src/partclone.h partclonebackup/src/partclone.h
--- partclone-0.2.69/src/partclone.h	2013-12-26 02:16:44.000000000 -0500
+++ partclonebackup/src/partclone.h	2014-05-26 19:55:52.647730049 -0400
@@ -129,6 +129,7 @@ extern void close_log();
 extern int io_all(int *fd, char *buffer, unsigned long long count, int do_write, cmd_opt *opt);
 extern void sync_data(int fd, cmd_opt* opt);
 extern void rescue_sector(int *fd, unsigned long long pos, char *buff, cmd_opt *opt);
+
 /**
  * for restore used functions
  * restore_image_hdr	- get image_head from image file 
diff -rupN partclone-0.2.69/src/progress.c partclonebackup/src/progress.c
--- partclone-0.2.69/src/progress.c	2013-12-26 02:16:44.000000000 -0500
+++ partclonebackup/src/progress.c	2014-05-26 19:55:52.647730049 -0400
@@ -184,6 +184,14 @@ static void calculate_speed(struct progr
 /// update information at progress bar
 extern void progress_update(struct progress_bar *prog, unsigned long long copied, unsigned long long current, int done)
 {
+	FILE *fog_log;
+	fog_log = fopen("/var/log/status.fog","w");
+	if (fog_log == NULL)
+	{
+		fprintf(stderr,"Error opening file %s.",fog_log);
+		exit(0);
+	}
+
     char clear_buf = ' ';
     prog_stat_t prog_stat;
 
@@ -202,6 +210,47 @@ extern void progress_update(struct progr
 	    fprintf(stderr, _(", %6.2f%s/min,"), prog_stat.speed, prog_stat.speed_unit);
 	if(prog->flag == IO)
 	    fprintf(stderr, "\n\r%80c\rcurrent block: %10lld, total block: %10lld, Complete: %6.2f%%%s\r", clear_buf, current, prog->total, prog_stat.total_percent, "\x1b[A");
+	double totalsize=prog->block_size * prog->total;
+	double currentsize=prog->block_size * current;
+	double kbyte=1024;
+	double mbyte=kbyte*kbyte;
+	double gbyte=mbyte*kbyte;
+	double tbyte=gbyte*kbyte;
+	double pbyte=tbyte*kbyte;
+	char current_str[15];
+	char total_str[15];
+	if (totalsize < kbyte)
+		sprintf(total_str,"%.2f B",totalsize);
+	if (totalsize >= kbyte)
+		sprintf(total_str,"%.2f KB",(totalsize/kbyte));
+	if (totalsize >= mbyte)
+		sprintf(total_str,"%.2f MB",(totalsize/mbyte));
+	if (totalsize >= gbyte)
+		sprintf(total_str,"%.2f GB",(totalsize/gbyte));
+	if (totalsize >= tbyte)
+		sprintf(total_str,"%.2f TB",(totalsize/tbyte));
+	if (totalsize >= pbyte)
+		sprintf(total_str,"%.2f PB",(totalsize/pbyte));
+	if (currentsize < kbyte)
+		sprintf(current_str,"%.2f B",currentsize);
+	if (currentsize >= kbyte)
+		sprintf(current_str,"%.2f KB",(currentsize/kbyte));
+	if (currentsize >= mbyte)
+		sprintf(current_str,"%.2f MB",(currentsize/mbyte));
+	if (currentsize >= gbyte)
+		sprintf(current_str,"%.2f GB",(currentsize/gbyte));
+	if (currentsize >= tbyte)
+		sprintf(current_str,"%.2f TB",(currentsize/tbyte));
+	if (currentsize >= pbyte)
+		sprintf(current_str,"%.2f PB",(currentsize/pbyte));
+	fprintf(fog_log,"%6.2f%s@",prog_stat.speed,prog_stat.speed_unit);
+	fprintf(fog_log,"%s@",prog_stat.Eformated);
+	fprintf(fog_log,"%s@",prog_stat.Rformated);
+	fprintf(fog_log,"%s@",current_str);
+	fprintf(fog_log,"%s@",total_str);
+	fprintf(fog_log,"%6.2f@",prog_stat.percent);
+	fprintf(fog_log,"%f\n",totalsize);
+
     } else {
 	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
@@ -218,12 +267,20 @@ extern void progress_update(struct progr
 	    fprintf(stderr, _("Ave. Rate: %6.1f%s/min, "), prog_stat.speed, prog_stat.speed_unit);
         fprintf(stderr, _("%s"), "100.00% completed!\n");
     }
+	fclose(fog_log);
 }
 
 /// update information at ncurses mode
 extern void Ncurses_progress_update(struct progress_bar *prog, unsigned long long copied, unsigned long long current, int done)
 {
 #ifdef HAVE_LIBNCURSESW
+    FILE *fog_log;
+    fog_log = fopen("/var/log/status.fog","w");
+    if (fog_log == NULL)
+    {
+        fprintf(stderr,"Error opening file %s.",fog_log);
+        exit(0);
+    }
 
     char *p_block, *t_block;
     prog_stat_t prog_stat;
@@ -275,10 +332,49 @@ extern void Ncurses_progress_update(stru
 
 	free(p_block);
 	free(t_block);
-        
 	wrefresh(p_win);
         wrefresh(bar_win);
         wrefresh(tbar_win);
+	double totalsize=prog->block_size * prog->total;
+	double currentsize=prog->block_size * current;
+	double kbyte=1024;
+	double mbyte=kbyte*kbyte;
+	double gbyte=mbyte*kbyte;
+	double tbyte=gbyte*kbyte;
+	double pbyte=tbyte*kbyte;
+	char current_str[15];
+	char total_str[15];
+	if (totalsize < kbyte)
+		sprintf(total_str,"%.2f B",totalsize);
+	if (totalsize >= kbyte)
+		sprintf(total_str,"%.2f KB",(totalsize/kbyte));
+	if (totalsize >= mbyte)
+		sprintf(total_str,"%.2f MB",(totalsize/mbyte));
+	if (totalsize >= gbyte)
+		sprintf(total_str,"%.2f GB",(totalsize/gbyte));
+	if (totalsize >= tbyte)
+		sprintf(total_str,"%.2f TB",(totalsize/tbyte));
+	if (totalsize >= pbyte)
+		sprintf(total_str,"%.2f PB",(totalsize/pbyte));
+	if (currentsize < kbyte)
+		sprintf(current_str,"%.2f B",currentsize);
+	if (currentsize >= kbyte)
+		sprintf(current_str,"%.2f KB",(currentsize/kbyte));
+	if (currentsize >= mbyte)
+		sprintf(current_str,"%.2f MB",(currentsize/mbyte));
+	if (currentsize >= gbyte)
+		sprintf(current_str,"%.2f GB",(currentsize/gbyte));
+	if (currentsize >= tbyte)
+		sprintf(current_str,"%.2f TB",(currentsize/tbyte));
+	if (currentsize >= pbyte)
+		sprintf(current_str,"%.2f PB",(currentsize/pbyte));
+    fprintf(fog_log,"%6.2f%s@",prog_stat.speed,prog_stat.speed_unit);
+	fprintf(fog_log,"%s@",prog_stat.Eformated);
+	fprintf(fog_log,"%s@",prog_stat.Rformated);
+	fprintf(fog_log,"%s@",current_str);
+	fprintf(fog_log,"%s@",total_str);
+	fprintf(fog_log,"%6.2f@",prog_stat.percent);
+	fprintf(fog_log,"%f\n",totalsize);
     } else {
         mvwprintw(p_win, 0, 0, _("Total Time: %s Remaining: %s "), prog_stat.Eformated, prog_stat.Rformated);
 	if((prog->flag == IO) || (prog->flag == NO_BLOCK_DETAIL))
@@ -308,6 +404,7 @@ extern void Ncurses_progress_update(stru
         refresh();
 	sleep(1);
     }
+	fclose(fog_log);
 
 #endif
 }
diff -rupN partclone-0.2.69/tests/Makefile partclonebackup/tests/Makefile
--- partclone-0.2.69/tests/Makefile	2013-12-26 02:16:44.000000000 -0500
+++ partclonebackup/tests/Makefile	1969-12-31 19:00:00.000000000 -0500
@@ -1,505 +0,0 @@
-# Makefile.in generated by automake 1.11.6 from Makefile.am.
-# tests/Makefile.  Generated from Makefile.in by configure.
-
-# Copyright (C) 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002,
-# 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011 Free Software
-# Foundation, Inc.
-# This Makefile.in is free software; the Free Software Foundation
-# gives unlimited permission to copy and/or distribute it,
-# with or without modifications, as long as this notice is preserved.
-
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY, to the extent permitted by law; without
-# even the implied warranty of MERCHANTABILITY or FITNESS FOR A
-# PARTICULAR PURPOSE.
-
-
-
-am__make_dryrun = \
-  { \
-    am__dry=no; \
-    case $$MAKEFLAGS in \
-      *\\[\ \	]*) \
-        echo 'am--echo: ; @echo "AM"  OK' | $(MAKE) -f - 2>/dev/null \
-          | grep '^AM OK$$' >/dev/null || am__dry=yes;; \
-      *) \
-        for am__flg in $$MAKEFLAGS; do \
-          case $$am__flg in \
-            *=*|--*) ;; \
-            *n*) am__dry=yes; break;; \
-          esac; \
-        done;; \
-    esac; \
-    test $$am__dry = yes; \
-  }
-pkgdatadir = $(datadir)/partclone
-pkgincludedir = $(includedir)/partclone
-pkglibdir = $(libdir)/partclone
-pkglibexecdir = $(libexecdir)/partclone
-am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
-install_sh_DATA = $(install_sh) -c -m 644
-install_sh_PROGRAM = $(install_sh) -c
-install_sh_SCRIPT = $(install_sh) -c
-INSTALL_HEADER = $(INSTALL_DATA)
-transform = $(program_transform_name)
-NORMAL_INSTALL = :
-PRE_INSTALL = :
-POST_INSTALL = :
-NORMAL_UNINSTALL = :
-PRE_UNINSTALL = :
-POST_UNINSTALL = :
-build_triplet = i686-pc-linux-gnu
-host_triplet = i686-pc-linux-gnu
-#am__append_1 = ext2.test \
-#	ext3.test ext4.test
-#am__append_2 = btrfs.test
-#am__append_3 = fat.test
-#am__append_4 = reiserfs.test
-
-#if ENABLE_REISER4
-#TESTS += reiser4.test
-#endif
-#am__append_5 = hfsplus.test
-#am__append_6 = xfs.test
-#am__append_7 = exfat.test
-#am__append_8 = ntfs.test
-
-#if ENABLE_UFS
-#TESTS += ufs.test
-#endif
-
-#if ENABLE_VMFS
-#TESTS += vmfs.test
-#endif
-#am__append_9 = jfs.test
-#am__append_10 = btrfs.test
-#am__append_11 = exfat.test
-#am__append_12 = ncursesw.test
-subdir = tests
-DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in TODO
-ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
-am__aclocal_m4_deps = $(top_srcdir)/m4/gettext.m4 \
-	$(top_srcdir)/m4/iconv.m4 $(top_srcdir)/m4/lib-ld.m4 \
-	$(top_srcdir)/m4/lib-link.m4 $(top_srcdir)/m4/lib-prefix.m4 \
-	$(top_srcdir)/m4/nls.m4 $(top_srcdir)/m4/po.m4 \
-	$(top_srcdir)/m4/progtest.m4 $(top_srcdir)/configure.ac
-am__configure_deps = $(am__aclocal_m4_deps) $(CONFIGURE_DEPENDENCIES) \
-	$(ACLOCAL_M4)
-mkinstalldirs = $(install_sh) -d
-CONFIG_HEADER = $(top_builddir)/config.h
-CONFIG_CLEAN_FILES =
-CONFIG_CLEAN_VPATH_FILES =
-SOURCES =
-DIST_SOURCES =
-am__can_run_installinfo = \
-  case $$AM_UPDATE_INFO_DIR in \
-    n|no|NO) false;; \
-    *) (install-info --version) >/dev/null 2>&1;; \
-  esac
-am__tty_colors = \
-red=; grn=; lgn=; blu=; std=
-DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-ACLOCAL = ${SHELL} /home/thomas/work_house/partclone/missing --run aclocal-1.11
-AMTAR = $${TAR-tar}
-AUTOCONF = ${SHELL} /home/thomas/work_house/partclone/missing --run autoconf
-AUTOHEADER = ${SHELL} /home/thomas/work_house/partclone/missing --run autoheader
-AUTOMAKE = ${SHELL} /home/thomas/work_house/partclone/missing --run automake-1.11
-AWK = gawk
-CC = gcc
-CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2 -Wall
-CPP = gcc -E
-CPPFLAGS = 
-CYGPATH_W = echo
-DEFS = -DHAVE_CONFIG_H
-DEPDIR = .deps
-ECHO_C = 
-ECHO_N = -n
-ECHO_T = 
-EGREP = /bin/grep -E
-EXEEXT = 
-GMSGFMT = /usr/bin/msgfmt
-GMSGFMT_015 = /usr/bin/msgfmt
-GREP = /bin/grep
-INSTALL = /usr/bin/install -c
-INSTALL_DATA = ${INSTALL} -m 644
-INSTALL_PROGRAM = ${INSTALL}
-INSTALL_SCRIPT = ${INSTALL}
-INSTALL_STRIP_PROGRAM = $(install_sh) -c -s
-INTLLIBS = 
-INTL_MACOSX_LIBS = 
-LDFLAGS = 
-LIBICONV = -liconv
-LIBINTL = 
-LIBOBJS = 
-LIBS = -lncursesw -lpthread 
-LN_S = ln -s
-LTLIBICONV = -liconv
-LTLIBINTL = 
-LTLIBOBJS = 
-MAKEINFO = ${SHELL} /home/thomas/work_house/partclone/missing --run makeinfo
-MKDIR_P = /bin/mkdir -p
-MSGFMT = /usr/bin/msgfmt
-MSGFMT_015 = /usr/bin/msgfmt
-MSGMERGE = /usr/bin/msgmerge
-OBJEXT = o
-ORIGINAL_CFLAGS = -g -O2
-PACKAGE = partclone
-PACKAGE_BUGREPORT = thomas@nchc.org.tw
-PACKAGE_NAME = Partclone
-PACKAGE_STRING = Partclone 0.2.69
-PACKAGE_TARNAME = partclone
-PACKAGE_URL = 
-PACKAGE_VERSION = 0.2.69
-PATH_SEPARATOR = :
-PKG_CONFIG = /usr/bin/pkg-config
-PKG_CONFIG_LIBDIR = 
-PKG_CONFIG_PATH = 
-POSUB = po
-RM = /bin/rm
-SET_MAKE = 
-SHELL = /bin/bash
-STRIP = 
-USE_NLS = yes
-UUID_CFLAGS = -I/usr/include/uuid  
-UUID_LIBS = -luuid  
-VERSION = 0.2.69
-XGETTEXT = /usr/bin/xgettext
-XGETTEXT_015 = /usr/bin/xgettext
-abs_builddir = /home/thomas/work_house/partclone/tests
-abs_srcdir = /home/thomas/work_house/partclone/tests
-abs_top_builddir = /home/thomas/work_house/partclone
-abs_top_srcdir = /home/thomas/work_house/partclone
-ac_ct_CC = gcc
-am__include = include
-am__leading_dot = .
-am__quote = 
-am__tar = $${TAR-tar} chof - "$$tardir"
-am__untar = $${TAR-tar} xf -
-bindir = ${exec_prefix}/bin
-build = i686-pc-linux-gnu
-build_alias = 
-build_cpu = i686
-build_os = linux-gnu
-build_vendor = pc
-builddir = .
-datadir = ${datarootdir}
-datarootdir = ${prefix}/share
-docdir = ${datarootdir}/doc/${PACKAGE_TARNAME}
-dvidir = ${docdir}
-exec_prefix = ${prefix}
-host = i686-pc-linux-gnu
-host_alias = 
-host_cpu = i686
-host_os = linux-gnu
-host_vendor = pc
-htmldir = ${docdir}
-includedir = ${prefix}/include
-infodir = ${datarootdir}/info
-install_sh = ${SHELL} /home/thomas/work_house/partclone/install-sh
-libdir = ${exec_prefix}/lib
-libexecdir = ${exec_prefix}/libexec
-localedir = ${datarootdir}/locale
-localstatedir = ${prefix}/var
-mandir = ${datarootdir}/man
-mkdir_p = /bin/mkdir -p
-oldincludedir = /usr/include
-pdfdir = ${docdir}
-prefix = /usr/local
-program_transform_name = s,x,x,
-psdir = ${docdir}
-sbindir = ${exec_prefix}/sbin
-sharedstatedir = ${prefix}/com
-srcdir = .
-sysconfdir = ${prefix}/etc
-target_alias = 
-top_build_prefix = ../
-top_builddir = ..
-top_srcdir = ..
-TESTS = dd.test $(am__append_1) $(am__append_2) $(am__append_3) \
-	$(am__append_4) $(am__append_5) $(am__append_6) \
-	$(am__append_7) $(am__append_8) $(am__append_9) \
-	$(am__append_10) $(am__append_11) $(am__append_12)
-CLEANFILES = floppy*
-all: all-am
-
-.SUFFIXES:
-$(srcdir)/Makefile.in:  $(srcdir)/Makefile.am  $(am__configure_deps)
-	@for dep in $?; do \
-	  case '$(am__configure_deps)' in \
-	    *$$dep*) \
-	      ( cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh ) \
-	        && { if test -f $@; then exit 0; else break; fi; }; \
-	      exit 1;; \
-	  esac; \
-	done; \
-	echo ' cd $(top_srcdir) && $(AUTOMAKE) --foreign tests/Makefile'; \
-	$(am__cd) $(top_srcdir) && \
-	  $(AUTOMAKE) --foreign tests/Makefile
-.PRECIOUS: Makefile
-Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
-	@case '$?' in \
-	  *config.status*) \
-	    cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh;; \
-	  *) \
-	    echo ' cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)'; \
-	    cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe);; \
-	esac;
-
-$(top_builddir)/config.status: $(top_srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
-	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
-
-$(top_srcdir)/configure:  $(am__configure_deps)
-	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
-$(ACLOCAL_M4):  $(am__aclocal_m4_deps)
-	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
-$(am__aclocal_m4_deps):
-tags: TAGS
-TAGS:
-
-ctags: CTAGS
-CTAGS:
-
-
-check-TESTS: $(TESTS)
-	@failed=0; all=0; xfail=0; xpass=0; skip=0; \
-	srcdir=$(srcdir); export srcdir; \
-	list=' $(TESTS) '; \
-	$(am__tty_colors); \
-	if test -n "$$list"; then \
-	  for tst in $$list; do \
-	    if test -f ./$$tst; then dir=./; \
-	    elif test -f $$tst; then dir=; \
-	    else dir="$(srcdir)/"; fi; \
-	    if $(TESTS_ENVIRONMENT) $${dir}$$tst; then \
-	      all=`expr $$all + 1`; \
-	      case " $(XFAIL_TESTS) " in \
-	      *[\ \	]$$tst[\ \	]*) \
-		xpass=`expr $$xpass + 1`; \
-		failed=`expr $$failed + 1`; \
-		col=$$red; res=XPASS; \
-	      ;; \
-	      *) \
-		col=$$grn; res=PASS; \
-	      ;; \
-	      esac; \
-	    elif test $$? -ne 77; then \
-	      all=`expr $$all + 1`; \
-	      case " $(XFAIL_TESTS) " in \
-	      *[\ \	]$$tst[\ \	]*) \
-		xfail=`expr $$xfail + 1`; \
-		col=$$lgn; res=XFAIL; \
-	      ;; \
-	      *) \
-		failed=`expr $$failed + 1`; \
-		col=$$red; res=FAIL; \
-	      ;; \
-	      esac; \
-	    else \
-	      skip=`expr $$skip + 1`; \
-	      col=$$blu; res=SKIP; \
-	    fi; \
-	    echo "$${col}$$res$${std}: $$tst"; \
-	  done; \
-	  if test "$$all" -eq 1; then \
-	    tests="test"; \
-	    All=""; \
-	  else \
-	    tests="tests"; \
-	    All="All "; \
-	  fi; \
-	  if test "$$failed" -eq 0; then \
-	    if test "$$xfail" -eq 0; then \
-	      banner="$$All$$all $$tests passed"; \
-	    else \
-	      if test "$$xfail" -eq 1; then failures=failure; else failures=failures; fi; \
-	      banner="$$All$$all $$tests behaved as expected ($$xfail expected $$failures)"; \
-	    fi; \
-	  else \
-	    if test "$$xpass" -eq 0; then \
-	      banner="$$failed of $$all $$tests failed"; \
-	    else \
-	      if test "$$xpass" -eq 1; then passes=pass; else passes=passes; fi; \
-	      banner="$$failed of $$all $$tests did not behave as expected ($$xpass unexpected $$passes)"; \
-	    fi; \
-	  fi; \
-	  dashes="$$banner"; \
-	  skipped=""; \
-	  if test "$$skip" -ne 0; then \
-	    if test "$$skip" -eq 1; then \
-	      skipped="($$skip test was not run)"; \
-	    else \
-	      skipped="($$skip tests were not run)"; \
-	    fi; \
-	    test `echo "$$skipped" | wc -c` -le `echo "$$banner" | wc -c` || \
-	      dashes="$$skipped"; \
-	  fi; \
-	  report=""; \
-	  if test "$$failed" -ne 0 && test -n "$(PACKAGE_BUGREPORT)"; then \
-	    report="Please report to $(PACKAGE_BUGREPORT)"; \
-	    test `echo "$$report" | wc -c` -le `echo "$$banner" | wc -c` || \
-	      dashes="$$report"; \
-	  fi; \
-	  dashes=`echo "$$dashes" | sed s/./=/g`; \
-	  if test "$$failed" -eq 0; then \
-	    col="$$grn"; \
-	  else \
-	    col="$$red"; \
-	  fi; \
-	  echo "$${col}$$dashes$${std}"; \
-	  echo "$${col}$$banner$${std}"; \
-	  test -z "$$skipped" || echo "$${col}$$skipped$${std}"; \
-	  test -z "$$report" || echo "$${col}$$report$${std}"; \
-	  echo "$${col}$$dashes$${std}"; \
-	  test "$$failed" -eq 0; \
-	else :; fi
-
-distdir: $(DISTFILES)
-	@srcdirstrip=`echo "$(srcdir)" | sed 's/[].[^$$\\*]/\\\\&/g'`; \
-	topsrcdirstrip=`echo "$(top_srcdir)" | sed 's/[].[^$$\\*]/\\\\&/g'`; \
-	list='$(DISTFILES)'; \
-	  dist_files=`for file in $$list; do echo $$file; done | \
-	  sed -e "s|^$$srcdirstrip/||;t" \
-	      -e "s|^$$topsrcdirstrip/|$(top_builddir)/|;t"`; \
-	case $$dist_files in \
-	  */*) $(MKDIR_P) `echo "$$dist_files" | \
-			   sed '/\//!d;s|^|$(distdir)/|;s,/[^/]*$$,,' | \
-			   sort -u` ;; \
-	esac; \
-	for file in $$dist_files; do \
-	  if test -f $$file || test -d $$file; then d=.; else d=$(srcdir); fi; \
-	  if test -d $$d/$$file; then \
-	    dir=`echo "/$$file" | sed -e 's,/[^/]*$$,,'`; \
-	    if test -d "$(distdir)/$$file"; then \
-	      find "$(distdir)/$$file" -type d ! -perm -700 -exec chmod u+rwx {} \;; \
-	    fi; \
-	    if test -d $(srcdir)/$$file && test $$d != $(srcdir); then \
-	      cp -fpR $(srcdir)/$$file "$(distdir)$$dir" || exit 1; \
-	      find "$(distdir)/$$file" -type d ! -perm -700 -exec chmod u+rwx {} \;; \
-	    fi; \
-	    cp -fpR $$d/$$file "$(distdir)$$dir" || exit 1; \
-	  else \
-	    test -f "$(distdir)/$$file" \
-	    || cp -p $$d/$$file "$(distdir)/$$file" \
-	    || exit 1; \
-	  fi; \
-	done
-check-am: all-am
-	$(MAKE) $(AM_MAKEFLAGS) check-TESTS
-check: check-am
-all-am: Makefile
-installdirs:
-install: install-am
-install-exec: install-exec-am
-install-data: install-data-am
-uninstall: uninstall-am
-
-install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
-
-installcheck: installcheck-am
-install-strip:
-	if test -z '$(STRIP)'; then \
-	  $(MAKE) $(AM_MAKEFLAGS) INSTALL_PROGRAM="$(INSTALL_STRIP_PROGRAM)" \
-	    install_sh_PROGRAM="$(INSTALL_STRIP_PROGRAM)" INSTALL_STRIP_FLAG=-s \
-	      install; \
-	else \
-	  $(MAKE) $(AM_MAKEFLAGS) INSTALL_PROGRAM="$(INSTALL_STRIP_PROGRAM)" \
-	    install_sh_PROGRAM="$(INSTALL_STRIP_PROGRAM)" INSTALL_STRIP_FLAG=-s \
-	    "INSTALL_PROGRAM_ENV=STRIPPROG='$(STRIP)'" install; \
-	fi
-mostlyclean-generic:
-
-clean-generic:
-	-test -z "$(CLEANFILES)" || rm -f $(CLEANFILES)
-
-distclean-generic:
-	-test -z "$(CONFIG_CLEAN_FILES)" || rm -f $(CONFIG_CLEAN_FILES)
-	-test . = "$(srcdir)" || test -z "$(CONFIG_CLEAN_VPATH_FILES)" || rm -f $(CONFIG_CLEAN_VPATH_FILES)
-
-maintainer-clean-generic:
-	@echo "This command is intended for maintainers to use"
-	@echo "it deletes files that may require special tools to rebuild."
-clean: clean-am
-
-clean-am: clean-generic mostlyclean-am
-
-distclean: distclean-am
-	-rm -f Makefile
-distclean-am: clean-am distclean-generic
-
-dvi: dvi-am
-
-dvi-am:
-
-html: html-am
-
-html-am:
-
-info: info-am
-
-info-am:
-
-install-data-am:
-
-install-dvi: install-dvi-am
-
-install-dvi-am:
-
-install-exec-am:
-
-install-html: install-html-am
-
-install-html-am:
-
-install-info: install-info-am
-
-install-info-am:
-
-install-man:
-
-install-pdf: install-pdf-am
-
-install-pdf-am:
-
-install-ps: install-ps-am
-
-install-ps-am:
-
-installcheck-am:
-
-maintainer-clean: maintainer-clean-am
-	-rm -f Makefile
-maintainer-clean-am: distclean-am maintainer-clean-generic
-
-mostlyclean: mostlyclean-am
-
-mostlyclean-am: mostlyclean-generic
-
-pdf: pdf-am
-
-pdf-am:
-
-ps: ps-am
-
-ps-am:
-
-uninstall-am:
-
-.MAKE: check-am install-am install-strip
-
-.PHONY: all all-am check check-TESTS check-am clean clean-generic \
-	distclean distclean-generic distdir dvi dvi-am html html-am \
-	info info-am install install-am install-data install-data-am \
-	install-dvi install-dvi-am install-exec install-exec-am \
-	install-html install-html-am install-info install-info-am \
-	install-man install-pdf install-pdf-am install-ps \
-	install-ps-am install-strip installcheck installcheck-am \
-	installdirs maintainer-clean maintainer-clean-generic \
-	mostlyclean mostlyclean-generic pdf pdf-am ps ps-am uninstall \
-	uninstall-am
-
-
-# Tell versions [3.59,3.63) of GNU make to not export all variables.
-# Otherwise a system limit (for SysV at least) may be exceeded.
-.NOEXPORT:
